configure.ac

```diff
@@ -304,36 +307,36 @@ CXXFLAGS+=" -DBOUT_VERSION_STRING=\"\\\"$BOUT_VERSION\\\"\" -DBOUT_VERSION_DOUBL
 #############################################################
 
 AS_IF([test "x$enable_backtrace" == "xyes" || test "x$enable_backtrace" == "xmaybe"], [
-    works=yes
-    AC_CHECK_PROG(works, addr2line, yes, no)
+    AC_CHECK_PROG([works], [addr2line], [yes], [no])
     AS_IF([test $works == no], [
-        AC_MSG_WARN("util addr2line not found")
+      AC_MSG_WARN("util addr2line not found")
     ])
-    AC_CHECK_FUNC(popen, , [works=no])
+    AC_CHECK_FUNC(popen, [works=yes], [works=no])
```

Comment:
this is a bug!
If addr2line is not working - we do not want to override the old result.

Reply:
Thanks, got distracted and pushed before I was done! Now bails out early if it doesn't find a component, which is what I started doing here

Reply:
Is there a reason to set works again to yes?
It should work now - I am just curious whether this a better style?

Reply:
No reason, I was just trying a few different ways of failing early and forgot to put it back to the original method. `AC_CHECK_FUNCS` (note the `S`) actually defines `HAVE_<func>` and I was initially emulating that, until I settled on the current method.